issue_4/contracts/GreenVault.sol

```diff
@@ -11,50 +22,82 @@ import "@yield-protocol/utils-v2/contracts/token/ERC20.sol";
 @notice A (single-token) vault that mints wrapped tokens
 in response to deposits
  */
-contract GreenVault is ERC20("GreenVaultToken", "OWL_GVT", 18) {
+contract GreenVault is
+    ERC20("GreenVaultToken", "OWL_GVT", 18),
+    Ownable,
+    FPMath
+{
     /**
     @notice the only token this vault supports
      */
-    IERC20 public theOnlySupportedToken;
+    IERC20Metadata public theOnlySupportedToken;
+
+    bool private is_entered;
+
+    /**
+    Multiplier: for each token deposited we mint `rewardFractionWad` Vault Tokens
+
+    It's represented as a wad (a 18-digit fixed point number)
+     */
+    uint256 public rewardFractionWad;
```

Comment:
I would call this an `exchangeRate`, since it gets applied in one direction when you deposit, but also in the opposite direction when you withdraw.